safenode/src/client/wallet.rs

```diff
@@ -73,14 +75,121 @@ impl SendClient for Client {
                 fee_ciphers,
             };
 
-            let _responses = self
+            let responses = self
                 .send_to_closest(Request::Cmd(cmd))
                 .await
                 .map_err(|err| Error::CouldNotSendTokens(err.to_string()))?;
 
-            // TODO: validate responses
+            // Get all Ok results of the expected response type `Spend`.
+            let ok_responses: Vec<_> = responses
+                .iter()
+                .flatten()
+                .flat_map(|resp| {
+                    if let Response::Cmd(CmdResponse::Spend(Ok(()))) = resp {
+                        Some(())
+                    } else {
+                        println!("Spend error {resp:?}.");
+                        None
+                    }
+                })
+                .collect();
+
+            // We require a majority of the close group to respond with Ok.
+            if ok_responses.len() >= close_group_majority() {
+                continue;
+            } else {
+                return Err(Error::CouldNotVerifyTransfer(format!(
+                    "Not enough close group nodes accepted the spend. Got {}, required: {}.",
+                    ok_responses.len(),
+                    close_group_majority()
+                )));
+            }
         }
 
         Ok(transfer)
     }
 }
+
+#[async_trait::async_trait]
+impl VerifyingClient for Client {
+    async fn verify(&self, dbc: &Dbc) -> Result<()> {
+        let mut received_spends = std::collections::BTreeSet::new();
+
+        for spend in &dbc.signed_spends {
+            let address = dbc_address(spend.dbc_id());
+            let query = Query::Spend(SpendQuery::GetDbcSpend(address));
+
+            let responses = self
+                .send_to_closest(Request::Query(query))
+                .await
+                .map_err(|err| Error::CouldNotVerifyTransfer(err.to_string()))?;
+
+            // Get all Ok results of the expected response type `GetDbcSpend`.
```

Comment:
this seems a code block (including the majority check part) with above?
also, seems for the `send_to_closest` , it doesn't need to wait_till_get_all_responses, but can return_early_when_get_majority_ok_responses, which can avoid a single no-response node hold back the whole process till time_out?

Reply:
Looking to see what I can do. Good catch, as above!

Reply:
Yes, this would need another low level api for client though, instead of send_to_closest (send_and_get_responses don't have the correct beahviour either).

Reply:
> seems for the send_to_closest , it doesn't need to wait_till_get_all_responses, but can return_early_when_get_majority_ok_responses,

Actually, we can't return early with majority of Ok responses, because we need majority of same item in those Ok responses.

Reply:
fixed